src/metrics/function_error.py

```diff
@@ -37,6 +37,28 @@ def evaluate(self, models: Iterable[ContextModel], num_batches: int = 1) -> Iter
 
         return errs
 
+class RegressionScore(Benchmark):
+    # zero_err is of shape (1)
+    # optimal_err is of shape (seq_length,)
+    def __init__(self, metric: Metric, function_class: FunctionClass, zero_err, optimal_err):
+        self.function_class = function_class
+        self.metric = metric
+        self.zero_err = zero_err
+        self.optimal_err = optimal_err.mean(dim = 1)
+
+    def evaluate(self, models: Iterable[ContextModel]) -> Iterable[Tensor]:
+        # generating model errors
+        model_funct = FunctionClassError(self.metric, self.function_class)
+        model_err = torch.Tensor(model_funct.evaluate(models))
+
+        # computations for reg_score
+        norm_model_err = torch.sub(model_err, self.zero_err)
+        norm_optimal_err = torch.sub(self.optimal_err, self.zero_err)
+
+        avg_model_err = torch.mean(norm_model_err, dim=(1,2))
+        avg_optimal_err = torch.mean(norm_optimal_err)
+
+        return avg_model_err/avg_optimal_err
```

Comment:
Divide the average model error *at each sequence length* by the optimal error at that index. This will produce an incorrect result

Reply:
Okay, but do we want a single value as the end score or a tensor of values?